lib/ansible/modules/system/cgroup.py

```diff
@@ -0,0 +1,256 @@
+'''
+
+EXAMPLES = '''
+# Checks a cgroup is present with only a memory limit
+- cgroup:
+    name: my_cgroup
+    state: present
+    memlimit: 256m
+
+# Checks a cgroup is absent
+- cgroup:
+    name: my_cgroup
+    state: absent
+'''
+
+RETURN = '''
+action:
+    description: If something changed the corresponding action (Creating, Updating or Deleting cgroup)
+    type: str
+    returned: only if changed
+config_diff:
+    description: If we're updating the configuration, the diff between the former and the new configuration
+    type: str
+    returned: only if updating configuration
+'''
+
+import os
+from ansible.module_utils.basic import AnsibleModule
+
+
+def run_module():
+    module_args = dict(
+        name=dict(type='str', required=True),
+        state=dict(type='str', required=True),
+        memlimit=dict(type='str', required=False),
+        memswaplimit=dict(type='str', required=False)
+    )
+
+    result = dict(
+        changed=False,
+        action='',
+        config_diff=''
+    )
+
+    module = AnsibleModule(
+        argument_spec=module_args,
+        supports_check_mode=False
+    )
+
+    name = module.params['name']
+    state = module.params['state']
+    memlimit = module.params['memlimit']
+    memswaplimit = module.params['memswaplimit']
+
+    config = {}
+    if memlimit is not None or memswaplimit is not None:
+        config['memory'] = {}
+
+    if memlimit is not None:
+        config['memory']['memory.limit_in_bytes'] = memlimit
+
+    if memswaplimit is not None:
+        config['memory']['memory.memsw.limit_in_bytes'] = memswaplimit
+
+    # Checks if cgroup exists
+    (found, raw_config) = read_config_from_file(name)
+
+    previous_state = 'present' if found else 'absent'
+    state_changed = state != previous_state
+
+    if state_changed:
+        result['action'] = ("Creating cgroup %s" if state == 'present' else "Deleting cgroup %s") % (name)
+
+    config_changed = False
+    if found and state != 'absent':
+        # Checks for changes in existing cgroup only if it exists and is to be kept
+        (config_changed, diff) = has_config_changed(raw_config, config)
+
+    if config_changed:
+        result['action'] = "Updating cgroup %s" % (name)
+        result['config_diff'] = diff
+
+    changed = state_changed or config_changed
+    result['changed'] = changed
+
+    # Check mode or nothing changed exiting here
+    if module.check_mode or not changed:
+        module.exit_json(**result)
+
+    # Something changed updating configuration
+    if state == 'present':
+        with open("/etc/cgconfig.d/%s.conf" % (name), 'a') as fh:
+            fh.truncate(0)
+            fh.write(generate_config_file_content(name, config))
+            fh.close()
+    else:
+        os.remove("/etc/cgconfig.d/%s.conf" % (name))
+
+    module.exit_json(**result)
+
+
+def read_config_from_file(desired_cgroup):
+    try:
+        with open("/etc/cgconfig.d/%s.conf" % (desired_cgroup), 'r') as fh:
```

Comment:
Should not the cgroup existence checked there ?

Reply:
Done